### ISSUE TITLE
YOLOv11 training speed

### ISSUE THREAD

**Issue by WYL-Projects** [ultralytics/ultralytics#19893]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I'm slow in training yolov11n-seg model with dual card 3090 24GB video memory. I am executing the following training commands:
`yolo train segment data=vstd_seg.yaml model=weights/yolov11_seg/yolo11n-seg.pt batch=64 device=0,1 imgsz=640 epochs=100 project=runs/segment/vstd_seg name=v10.0 workers=8`

The training process is shown below, from the training process, one epoch takes 1.5 hours which is very slow. In addition, when there is a lag during the training process, one of the two graphics cards will have a utilization of 0. I don't know what causes this, is it caused by the amount of data?
![Image](https://github.com/user-attachments/assets/e24ba600-f125-46eb-a0bc-4ac39220900f)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @WYL-Projects, thank you for bringing this to our attention and for your interest in Ultralytics 🚀! We understand how important training speed and efficiency are for your projects. 

We recommend checking out the [Docs](https://docs.ultralytics.com/) for general guidance and tips. For example, you can review the [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples to ensure optimal configurations.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us investigate further. For performance-related issues, details like dataset size, image resolution, and any custom modifications to the model or training settings are helpful.

To ensure you are running the latest features and fixes, please upgrade to the latest version of the `ultralytics` package, along with all required dependencies, in a Python>=3.8 environment:

```bash
pip install -U ultralytics
```

Additionally, consider the following:
- Ensure your GPUs are properly utilized and configured. You can validate this by checking your CUDA setup and using tools like `nvidia-smi` to monitor GPU usage.
- Try reducing the `batch` size or adjusting the `workers` value to better balance GPU and CPU utilization.

For further assistance, you can join the Ultralytics community:
- [Discord](https://discord.com/invite/ultralytics) for real-time chat 🎧
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and explore ideas with the community

## Verified Environments

YOLO can be run in any of the following environments, which are pre-configured with all necessary dependencies:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing, confirming correct operation of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across platforms.

This is an automated response to give you a starting point. An Ultralytics engineer will review your issue and provide further assistance soon 😊.

**Comment by Y-T-G** [comment]
Can you provide the output after running this command in terminal: `yolo checks`?

**Comment by WYL-Projects** [comment]
@Y-T-G One pattern I've noticed is that the progress bar for training increases by the number num_workers each time and then gets stuck for 3~4 seconds.

**Comment by Blase0707** [comment]
@WYL-Projects @Y-T-G I’m experiencing a similar issue while training the yolov8n model using a Tesla V100 32GB GPU. The progress bar for training increases by the number of `num_workers` each time and then stuck for several seconds. This behavior persists even after adjusting the `num_workers` and `batch`, or setting the `cache` option to "disk". The training code is as follows: 
```
from ultralytics import YOLO
model = YOLO('ultralytics/cfg/models/v8/yolov8n.yaml').load('yolov8n.pt')
model.train(data='ultralytics/cfg/datasets/coco.yaml', epochs=100, imgsz=640, save_period=5, device=0, name='train', batch=32, workers=8, cache='disk')
```

The results from running `yolo checks` in the terminal:
```
Ultralytics 8.3.99 🚀 Python-3.8.20 torch-1.12.1 CUDA:0 (Tesla V100-DGXS-32GB, 32475MiB)
Setup complete ✅ (40 CPUs, 251.8 GB RAM, 1527.5/1759.0 GB disk)

OS                  Linux-4.15.0-55-generic-x86_64-with-glibc2.17
Environment         Docker
Python              3.8.20
Install             git
Path                /workspace/Share/Oscar/ultralytics/ultralytics
RAM                 251.82 GB
Disk                1527.5/1759.0 GB
CPU                 Intel Xeon E5-2698 v4 2.20GHz
CPU count           40
GPU                 Tesla V100-DGXS-32GB, 32475MiB
GPU count           4
CUDA                10.2

numpy               ✅ 1.24.3<=2.1.1,>=1.23.0
matplotlib          ✅ 3.7.5>=3.3.0
opencv-python       ✅ 4.11.0.86>=4.6.0
pillow              ✅ 10.4.0>=7.1.2
pyyaml              ✅ 6.0.2>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.10.1>=1.4.1
torch               ✅ 1.12.1>=1.8.0
torchvision         ✅ 0.13.1>=0.9.0
tqdm                ✅ 4.67.1>=4.64.0
psutil              ✅ 7.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.0.3>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.14>=2.0.0
torch               ✅ 1.12.1!=2.4.0,>=1.8.0; sys_platform == "win32"
```

**Comment by rsy980** [comment]
多卡训练时，DataLoader 获取数据时卡住了，在获取下一批数据时发生了阻塞，将workers设置为0

**Comment by glenn-jocher** [comment]
This appears to be a data loading bottleneck issue that's common when training on multiple GPUs. When the progress bar increases by the number of workers and then stalls, it typically indicates the DataLoader is struggling to prepare batches fast enough for your GPUs.

Some solutions to try:

1. Set `workers=0` as @rsy980 suggested to eliminate DataLoader parallelism issues, though this may not be optimal for performance
2. Try increasing cache usage with `cache=True` to keep more data in memory
3. Check your dataset storage - if it's on a slow disk or network drive, consider moving it to local SSD
4. Examine your dataset preprocessing steps for potential bottlenecks

You could also try setting `pin_memory=True` in your training command to speed up CPU to GPU memory transfers. If none of these solutions work, could you please share your dataset size (number of images and their resolution) and the complete output of `yolo checks`?
